disnake/message.py

```diff
@@ -321,8 +323,25 @@ def __init__(self, *, data: AttachmentPayload, state: ConnectionState) -> None:
         self.waveform: Optional[bytes] = (
             b64decode(waveform_data) if (waveform_data := data.get("waveform")) else None
         )
+        _params = urlparse(self.url)
+        self._ex = parse_qs(_params.query)["ex"][0]
```

Comment:
Is this always going to exist? I can imagine this could raise a key error

Reply:
True it's a error of mine (the funny thing is that i even considered this previously but forgot to put it in code), the new attachment url parameters will be present only for links of attachments uploaded after the new cdn update (which is already active) however old attachment urls will not have these parameters. The details on how the cdn will treat old attachments is still unknown, though they will undergo to the same expiration behaviour of newer attachments.